src/app/modules/issue/issue.service.ts

```diff
@@ -0,0 +1,182 @@
+import { ObjectId } from 'bson'
+import mongoose from 'mongoose'
+import { errAsync, okAsync, ResultAsync } from 'neverthrow'
+
+import { IFormIssueSchema, IPopulatedForm } from 'src/types'
+
+import { createLoggerWithLabel } from '../../config/logger'
+import getFormIssueModel from '../../models/form_issue.server.model'
+import {
+  MailGenerationError,
+  MailSendError,
+} from '../../services/mail/mail.errors'
+import MailService from '../../services/mail/mail.service'
+import { getStartOfDay } from '../../utils/date'
+import { DatabaseError } from '../core/core.errors'
+import { FormNotFoundError } from '../form/form.errors'
+
+const logger = createLoggerWithLabel(module)
+
+const FormIssueModel = getFormIssueModel(mongoose)
+
+/**
+ * Inserts given form issue to the database.
+ * @param formId the formId to insert the feedback for
+ * @param issue the description of the issue reported
+ * @param email the contact info for form creator to follow up the issue with
+ * @returns ok(true) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+export const insertFormIssue = ({
+  formId,
+  issue,
+  email,
+}: {
+  formId: string
+  issue: string
+  email?: string
+}): ResultAsync<IFormIssueSchema, FormNotFoundError | DatabaseError> => {
+  if (!mongoose.Types.ObjectId.isValid(formId)) {
+    return errAsync(new FormNotFoundError())
+  }
+  return ResultAsync.fromPromise(
+    FormIssueModel.create({
+      formId: formId,
+      issue: issue,
+      ...(email && { email: email }),
```

Comment:
/nit can be terser by using Object shorthand as well

Reply:
done